### PR TITLE
test(#177): filter rooms dusk tests

### PR DESCRIPTION
**Pull request by jonathanp16**
##### _List all issues this PR resolves_
- #177 
- #428 
- #587 
-  #588 

##### _Complete these items before requesting reviews_
- [x] Static Analysis Rules Validated
- [x] All commits follow naming conventions

### Sub-tasks Checklist
- [ ] UI Implementation
- [x] UI Test
- [ ] Backend Implementation
- [ ] Backend Tests
- [ ] Integration Tests
- [ ] System Tests
- [x] Documentation

### CODE REVIEW

**Comment by EvanDime** [comment]
@jonathanp16 Is this ready for review?

File: tests/Browser/FilterRoomsDuskTest.php [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+            $browser
+                ->assertSourceHas('<title>CSU Booking Platform</title>');
+
+            $browser->press("@toggle-advanced-filters");
+            $browser->assertSourceHas('<h2>Capacity Standing</h2>');
+
+            $browser->assertSee('Art Nook');
+            $browser->type('#sofas', 1);
+            $browser->press("#filter-rooms");
+            $browser->pause(2000);
```
Comment (alexstojda) [inline tests/Browser/FilterRoomsDuskTest.php:63]:
```suggestion
            $browser->pause(250);
```

2 seconds is probably longer than needed

File: tests/Browser/FilterRoomsDuskTest.php [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+            $browser->assertSourceMissing('Art Nook');
+            $browser->script('location.reload();');
+
+            $browser->assertSee('Art Nook');
+            $browser->press("@toggle-advanced-filters");
+            $browser->check('@alcohol');
+            $browser->press("#filter-rooms");
+            $browser->pause(2000);
```
Comment (alexstojda) [inline tests/Browser/FilterRoomsDuskTest.php:71]:
```suggestion
            $browser->pause(250);
```

File: tests/Browser/FilterRoomsDuskTest.php [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+                ->assertSourceHas('<title>CSU Booking Platform</title>');
+
+            $browser->press("@toggle-advanced-filters");
+            $browser->assertSourceHas('<h2>Capacity Standing</h2>');
+
+            $browser->press('#filter-add-date');
+
+            $browser->within( new DateTimePicker('start_time-0'), function($browser) {
+                $browser->setDatetime(10,13);
+            })->pause(1000);
```
Comment (alexstojda) [inline tests/Browser/FilterRoomsDuskTest.php:135]:
```suggestion
            })->pause(250);
```

File: tests/Browser/FilterRoomsDuskTest.php [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+
+            $browser->within( new DateTimePicker('end_time-0'), function($browser) {
+                $browser->setDatetime(10,14);
+            })->pause(1000);
```
Comment (alexstojda) [inline tests/Browser/FilterRoomsDuskTest.php:139]:
```suggestion
            })->pause(250);
```

**Review by alexstojda — CHANGES_REQUESTED** [review]
Every pause adds unnecessary delays to the CI pipeline! Keep them as short as possible

**Review by EvanDime — DISMISSED** [review]
I agree with alex about the pauses, even though i was the first one to put them there. My computer is really slow so it does need the pause.

File: tests/Browser/FilterRoomsDuskTest.php (hunk 1 above)
Comment (jonathanp16) [reply tests/Browser/FilterRoomsDuskTest.php]: Fixed

File: tests/Browser/FilterRoomsDuskTest.php (hunk 2 above)
Comment (jonathanp16) [reply tests/Browser/FilterRoomsDuskTest.php]: Fixed

File: tests/Browser/FilterRoomsDuskTest.php (hunk 3 above)
Comment (jonathanp16) [reply tests/Browser/FilterRoomsDuskTest.php]: Fixed

File: tests/Browser/FilterRoomsDuskTest.php (hunk 4 above)
Comment (jonathanp16) [reply tests/Browser/FilterRoomsDuskTest.php]: Fixed

**Comment by jonathanp16** [comment]
Need to refactor these due to changes in the UI

**Review by EvanDime — APPROVED** [review]

**Review by alexstojda — APPROVED** [review]

**Pull request merged**